GoogleTestAdapter/Core/Runners/CommandLineGenerator.cs

```diff
@@ -69,39 +69,59 @@ private IEnumerable<Args> GetFinalCommandLines(string baseCommandLine)
                 return commandLines;
             }
 
-            string baseCommandLineWithFilter = baseCommandLine + GoogleTestConstants.FilterOption;
-
             List<string> suitesRunningAllTests = GetSuitesRunningAllTests();
-            baseCommandLineWithFilter += GetFilterForSuitesRunningAllTests(suitesRunningAllTests);
-
-            List<TestCase> testCasesNotRunBySuite = GetTestCasesNotRunBySuite(suitesRunningAllTests);
-            List<TestCase> testCasesRunBySuite = _testCasesToRun.Where(tc => !testCasesNotRunBySuite.Contains(tc)).ToList();
-            if (testCasesNotRunBySuite.Count == 0)
+            int suiteLengthAggregate = 0;
+            int groupIdxCounter = 0;
+            int maxSuiteLength = MaxCommandLength - _lengthOfExecutableString - userParam.Length - 1;
+
+            var suiteGroups = suitesRunningAllTests
+                              .Select(suite =>
+                                { suiteLengthAggregate += suite.Length + 3;
+                                  if (suiteLengthAggregate > maxSuiteLength) {
+                                      ++groupIdxCounter;
+                                      suiteLengthAggregate = suite.Length + 3;
+                                  }
+                                  return new { groupIdx = groupIdxCounter, suiteName = suite };                              
+                                })
+                              .GroupBy(pair => pair.groupIdx, pair => pair.suiteName)
+                              .ToList();
+
+            if (!suiteGroups.Any()) suiteGroups = Enumerable.Range(-1, 1).GroupBy(dummy => dummy, dummy => string.Empty).ToList();
```

Comment:
What's this? Don't we just need to proceed to test case handling if we do not have any suiteGroups, i.e., we do not have any suites at all?

Reply:
This is only a trick so I don't need no (or nearly no) special handling for this case but just can iterater "normally" over the suiteGroups (i.e. without an entry in suiteGroups nothing would be processed).

Reply:
Ok, I see - my confusion was caused by assuming that the `Enumerable.Range()` parameters were lower and upper bound (not all of us happen to be Linq wizards ;-) ). However, I think there's a better option:
* just collect the suite groups as you already do
* iterate over the first `n-1` of them, basically performing your `if` case
* then perform your `else` case (outside the loop)

There's some tiny code duplication from the pre-`if` part of the loop (which could be reduced e.g. by introducing method `GetTestCasesOfSuiteGroup()`, which might indeed be a good idea), but I think it's easier to understand (especially after extracting the method as suggested at line 77, since no knowledge of your trick is necessary at the call side).

Reply:
Your proposed approach would not help in avoiding the 'trick', because the trick is not about special handling for the last groupSuite but about having a groupSuite at all. The trick would be much more elegant if i could create the 'fake'-grouping which results in an empty string-list if ToList() is applied on therefore the `suiteGroup.Key >= 0` clause would be unneccessary (but I failed to do so and I did not want to spend to much time on this).

And yes I could seperate things as you propose, but I would need to extract the "preamble" (before the `if (suiteGroup != suiteGroups.Last())`) into a method (or lambda) to avoid code-duplication. Not sure if this improves readability/coherence a lot, but I will give it a try (however not before monday - because I don't have the right VisualStudio version installed here)